lib/parsers/exif_parser.rb

```diff
@@ -167,10 +174,36 @@ def method_missing(*a)
   # Squash exifr's invalid date warning since we do not use that data.
   EXIFR.logger = Logger.new(nil)
 
-  def exif_from_tiff_io(constrained_io)
+  def exif_from_tiff_io(constrained_io, should_include_sub_ifds = false)
     Measurometer.instrument('format_parser.EXIFParser.exif_from_tiff_io') do
-      raw_exif_data = EXIFR::TIFF.new(IOExt.new(constrained_io))
-      raw_exif_data ? EXIFResult.new(raw_exif_data) : nil
+      extended_io = IOExt.new(constrained_io)
+      raw_exif_data = EXIFR::TIFF.new(extended_io)
+      sub_ifds_hash = {}
+
+      if should_include_sub_ifds
+        sub_ifds_offsets = raw_exif_data.map(&:sub_ifds).flatten.compact
+        sub_ifds_hash = load_sub_ifds(extended_io, sub_ifds_offsets)
+      end
+
+      return raw_exif_data ? EXIFResult.new(raw_exif_data, sub_ifds_hash) : nil
+    end
+  end
+
+  private
+
+  # Reads exif data from subIFDs. This is important for NEF files.
+  def load_sub_ifds(extended_io, sub_ifds_offsets)
+    # Returning an hash of subIFDs using offsets as keys
+    # {
+    #    123 => { subIFD data...}
+    #    456 => { another subIFD data...}
+    # }
+    unless sub_ifds_offsets.empty?
```

Comment:
To get rid of a level of indentation, you can just return early: `return if sub_ifds_offsets.empty?`.